www/data_service/src/services/stream/stream.service.coffee

```diff
@@ -0,0 +1,36 @@
+class Stream extends Factory
+    constructor: ($log) ->
+        return class StreamInstance
+            # the unsubscribe listener will be called on each unsubscribe call
+            onUnsubscribe: null
+
+            constructor: ->
+                @listeners = []
+
+            subscribe: (listener) ->
+                if not angular.isFunction(listener)
+                    $log.debug("#{listener} is not a function")
```

Comment:
Maybe we should throw an error here? This interface is likely to be used only by ourselves, we may not let it pass silently(though there is a log). Instead an error stop the program as soon as possible is better.